src/components/button/index.tsx

```diff
@@ -103,21 +103,32 @@ class Button extends PureComponent<Props, ButtonState> {
   }
 
   getLabelColor() {
-    const {linkColor, outline, outlineColor, disabled, color: propsColor, backgroundColor, modifiers} = this.props;
+    const {
+      linkColor,
+      fullWidth,
+      outline,
+      outlineColor,
+      disabled,
+      color: propsColor,
+      backgroundColor,
+      modifiers
+    } = this.props;
     const {color: modifiersColor} = modifiers;
     const isLink = this.isLink;
+    const contentColor = Colors.getRelativeColor(this.getBackgroundColor());
+
+    let color: string | undefined = contentColor;
 
-    let color: string | undefined = Colors.$textDefaultLight;
     if (isLink) {
-      color = linkColor || Colors.$textPrimary;
+      color = linkColor || contentColor || Colors.$textPrimary;
     } else if (outline) {
-      color = outlineColor || Colors.$textPrimary;
+      color = outlineColor || contentColor || Colors.$textPrimary;
     } else if (this.isIconButton) {
-      color = backgroundColor === 'transparent' ? undefined : Colors.$iconDefaultLight;
+      color = backgroundColor === 'transparent' ? undefined : contentColor || Colors.$textPrimary;
     }
 
-    if (disabled && (isLink || outline)) {
-      return Colors.$textDisabled;
+    if (disabled) {
+      return isLink || outline || fullWidth ? Colors.$textDisabled : Colors.white;
```

Comment:
Is this correct? The color of the disabled button is white?

Reply:
I'm wondering about disabled color in design kits and dark mode. Is it always white?